src/storage/v3/shard_rsm.cpp

```diff
@@ -457,6 +492,129 @@ EdgeFiller InitializeEdgeFillerFunction(const msgs::ExpandOneRequest &req) {
   return edge_filler;
 }
 
+msgs::ReadResponses HandleReadWithOrderBy(msgs::ExpandOneRequest &&req, Shard::Accessor &&acc) {
+  bool action_successful = true;
+
+  std::vector<msgs::ExpandOneResultRow> results;
+  auto batch_limit = req.limit;
+  auto dba = DbAccessor{&acc};
+
+  auto maybe_filter_based_on_edge_uniquness = InitializeEdgeUniqunessFunction(req.only_unique_neighbor_rows);
+  auto edge_filler = InitializeEdgeFillerFunction(req);
+
+  auto vertex_iterable = acc.Vertices(View::OLD);
+  const auto ordered_vertices = OrderByElements<VertexAccessor>(acc, dba, vertex_iterable, req.order_by);
+
+  for (const auto &ordered_vertice : ordered_vertices) {
+    // Get Vertex acc
+    auto src_vertex_acc = ordered_vertice.object_acc;
+
+    if (!req.filters.empty()) {
+      // NOTE - DbAccessor might get removed in the future.
+      const bool eval = FilterOnVertex(dba, src_vertex_acc, req.filters, expr::identifier_node_symbol);
+      if (!eval) {
+        continue;
+      }
+    }
+
+    auto [in_edge_accessors, out_edge_accessors] = GetEdgesFromVertex(src_vertex_acc, req.direction);
+    const auto in_ordered_edges = OrderByElements<EdgeAccessor>(acc, dba, in_edge_accessors, req.order_by);
+    const auto out_ordered_edges = OrderByElements<EdgeAccessor>(acc, dba, out_edge_accessors, req.order_by);
+
+    std::vector<EdgeAccessor> in_edge_ordered_accessors;
+    std::transform(in_ordered_edges.begin(), in_ordered_edges.end(), std::back_inserter(in_edge_ordered_accessors),
+                   [](auto &edge_element) { return edge_element.object_acc; });
+
+    std::vector<EdgeAccessor> out_edge_ordered_accessors;
+    std::transform(out_ordered_edges.begin(), out_ordered_edges.end(), std::back_inserter(out_edge_ordered_accessors),
+                   [](auto &edge_element) { return edge_element.object_acc; });
+
+    auto label_id = src_vertex_acc.PrimaryLabel(View::NEW);
+    if (label_id.HasError()) {
+      action_successful = false;
+      break;
+    }
+    auto primary_key = src_vertex_acc.PrimaryKey(View::NEW);
+    if (primary_key.HasError()) {
+      action_successful = false;
+      break;
+    }
+
+    msgs::VertexId src_vertice(msgs::Label{.id = *label_id}, conversions::ConvertValueVector(*primary_key));
```

Comment:
`src_vertice` is also a weird name.

Reply:
Agree, I would still keep it like this. Reason is that `ExpandOneRequest` has member `src_vertices` and I believe we need to remain consistent.
Alternatively, I can modify `ExpandOneRequest` as well and give a different name to both